lib/Compressor.js

```diff
@@ -159,9 +186,18 @@ export class Compressor extends Transformer {
   }
 
   _transformTypedValue({entries, termType, value, termInfo}) {
-    const EncoderClass = TYPE_ENCODERS.get(termType);
-    const encoder = EncoderClass && EncoderClass.createEncoder(
-      {value, transformer: this, termInfo});
+    let encoder;
+
+    encoder = TypedLiteralEncoder.createEncoder({
+      value,
+      transformer: this,
+      termType
+    });
+    if(!encoder) {
+      const EncoderClass = TYPE_ENCODERS.get(termType);
+      encoder = EncoderClass && EncoderClass.createEncoder(
+        {value, transformer: this, termInfo});
+    }
```

Comment:
We could get crazier and do this:

```suggestion
    const encoder = TypedLiteralEncoder.createEncoder({
      value,
      transformer: this,
      termType
    }) || TYPE_ENCODERS.get(termType)?.createEncoder({
      value,
      transformer: this,
      termInfo
    });
```